BombsExcersice/BombsExcersice.cpp

```diff
@@ -1,10 +1,173 @@
-// BombsExcersice.cpp : This file contains the 'main' function. Program execution begins and ends there.
-//
-
 #include <iostream>
+#include <vector>
+#include <fstream>
+#include <string>
+
+enum Type {Empty, Bomb, Wall};
+
+class Cell
+{
+public:
+    Cell() : type{ Empty }, lifetime{ 0 }, maxLifetime{ 3 } {}
+
+    void setType(Type t) { type = t; }
+    Type getType() const { return type; }
+
+    void becomeEmpty()
+    {
+        type = Empty;
+        lifetime = 0;
+    }
+    
+    int getLifetime() const { return lifetime; }
+    int getMaxLifetime() const { return maxLifetime; }
+    void increaseLifetime() { ++lifetime; }
+
+private:
+    Type type;
+    int lifetime;
+    int maxLifetime;
+};
+
+class BombsField
+{
+public:
+    BombsField(const int& rows, const int& cols) : rows(rows), cols(cols), field(rows, std::vector<Cell>(cols)) {}
+
+    void print() const
+    {
+        std::ofstream outputFile("output.txt");
+
+        for (int i = 0; i < rows; ++i)
+        {
+            for (int j = 0; j < cols; ++j)
+            {   
+                switch (field[i][j].getType())
+                {
+                    case Empty: outputFile << '.'; break;
+                    case Bomb: outputFile << 'O'; break;
+                    case Wall: outputFile << '#'; break;
+                }
+            }
+            outputFile << std::endl;
+        }
+        outputFile.close();
+    }
+
+    void setBomb(int x, int y) { field[x][y].setType(Bomb); }
```

Comment:
These two functions could be combined into a generic one: `void setCellType(Type type, int x, int y)`. Or you can even delete them and simply use `field[x][y].setType(type);` when it's needed. This allows you not to add an extra function each time you add a new cell type, and is more flexible when it comes to initializing the field.